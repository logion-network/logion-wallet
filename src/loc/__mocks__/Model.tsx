import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
