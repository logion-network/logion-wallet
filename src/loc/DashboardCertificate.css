.DashboardCertificate .frame-container {
    margin-top: 30px;
}

.DashboardCertificate pre {
    font-size: 1rem;
    font-family: "Montserrat", sans-serif;
    line-break: anywhere;
    white-space: break-spaces;
}
