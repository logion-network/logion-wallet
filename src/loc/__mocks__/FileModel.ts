export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
