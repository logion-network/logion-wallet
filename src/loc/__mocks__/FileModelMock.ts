export let getAllDeliveries = jest.fn().mockResolvedValue({});
