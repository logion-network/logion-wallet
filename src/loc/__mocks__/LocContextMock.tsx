import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
