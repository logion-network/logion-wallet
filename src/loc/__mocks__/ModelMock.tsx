export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
