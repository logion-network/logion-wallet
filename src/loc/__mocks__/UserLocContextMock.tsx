import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
