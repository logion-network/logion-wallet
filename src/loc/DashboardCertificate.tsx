import { CheckCertifiedCopyResult, CheckHashResult } from "@logion/client";
import { useEffect, useMemo, useState, useCallback } from "react";
import { useParams } from "react-router-dom";

import { useCommonContext } from "src/common/CommonContext";
import PolkadotFrame from "src/common/PolkadotFrame";
import CheckDeliveredFrame from "src/components/deliverycheck/CheckDeliveredFrame";
import ItemFiles from "src/components/itemfiles/ItemFiles";
import { useLogionChain } from "src/logion-chain";
import { getAllDeliveries, ItemDeliveriesResponse } from "./FileModel";
import { useLocContext } from "./LocContext";
import LocPane from "./LocPane";
import { CertificateItemDetails } from "src/components/certificateitemdetails/CertificateItemDetails";
import FrameTitle from "src/components/frametitle/FrameTitle";

import "./DashboardCertificate.css";
import CheckFileFrame from "../components/checkfileframe/CheckFileFrame";

export default function DashboardCertificate() {
    const itemId = useParams<"itemId">().itemId;
    const {
        backPath,
        locId,
        loc,
        collectionItems,
    } = useLocContext();
    const collectionItem = useMemo(() => collectionItems.find(item => item.id === itemId), [ itemId, collectionItems ]);
    const [ deliveries, setDeliveries ] = useState<ItemDeliveriesResponse>();
    const { axiosFactory } = useLogionChain();
    const { colorTheme } = useCommonContext();
    const [ checkCertifiedCopyResult, setCheckCertifiedCopyResult ] = useState<CheckCertifiedCopyResult>();
    const [ checkHashResult, setCheckHashResult ] = useState<CheckHashResult>();

    useEffect(() => {
        if(loc && itemId && deliveries === undefined) {
            (async function() {
                const deliveries = await getAllDeliveries(axiosFactory!(loc?.ownerAddress), { locId: locId.toString(), collectionItemId: itemId });
                setDeliveries(deliveries);
            })();
        }
    }, [ collectionItem, axiosFactory, itemId, locId, loc, deliveries ]);

    const checkHash = useCallback((hash: string) => {
        if (collectionItem) {
            const result = collectionItem.checkHash(hash);
            setCheckHashResult(result);
        }
    }, [ collectionItem ]);

    if(!collectionItem || !loc) {
        return null;
    }

    return (
        <LocPane
            backPath={ backPath }
            loc={ loc }
            className="DashboardCertificate"
        >
            <PolkadotFrame
                title={
                    <FrameTitle iconId="polkadot_collection" text="Collection Item recorded data"/>
                }
            >
                <CertificateItemDetails
                    item={ collectionItem }
                />
            </PolkadotFrame>
            {
                collectionItem.files.length > 0 &&
                <>
                <div className="frame-container">
                    <PolkadotFrame>
                        <ItemFiles
                            collectionLoc={ loc }
                            item={ collectionItem }
                            deliveries={ deliveries }
                            checkCertifiedCopyResultResult={ checkCertifiedCopyResult }
                            checkHashResult={ checkHashResult }
                        />
                    </PolkadotFrame>
                </div>
                <div className="frame-container">
                    <CheckFileFrame
                        checkHash={ checkHash }
                        checkResult={ checkHashResult === undefined ? "NONE" : ( checkHashResult.collectionItemFile ? "POSITIVE" : "NEGATIVE") }
                        context="Collection Item"
                        checkedItem="file"
                    />
                </div>
                <div className="frame-container">
                    <CheckDeliveredFrame
                        item={ collectionItem }
                        colorTheme={ colorTheme }
                        detailedError={ true }
                        onChecked={ setCheckCertifiedCopyResult }
                    />
                </div>
                </>
            }
        </LocPane>
    );
}
