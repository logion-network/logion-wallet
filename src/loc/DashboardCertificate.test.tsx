import { CollectionItem } from "@logion/client";
import { shallowRender } from "src/tests";
import { setParams } from "src/__mocks__/ReactRouterMock";
import DashboardCertificate from "./DashboardCertificate";
import { setCollectionItems } from "./__mocks__/LocContextMock";
import { Hash } from "@logion/node-api";

jest.mock("./LocContext");

describe("DashboardCertificate", () => {

    it("renders empty", () => {
        const element = shallowRender(<DashboardCertificate />);
        expect(element).toMatchSnapshot();
    });

    it("renders item", () => {
        const itemId = Hash.of("some-id");
        setParams({ itemId: itemId.toHex() });
        setCollectionItems([
            {
                id: itemId,
                files: [],
            } as unknown as CollectionItem
        ])
        const element = shallowRender(<DashboardCertificate />);
        expect(element).toMatchSnapshot();
    });
});
