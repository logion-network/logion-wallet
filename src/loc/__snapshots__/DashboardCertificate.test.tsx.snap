// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCertificate renders empty 1`] = `null`;

exports[`DashboardCertificate renders item 1`] = `
<LocPane
  className="DashboardCertificate"
  loc={
    Object {
      "id": UUID {
        "bytes": Uint8Array [
          174,
          212,
          198,
          228,
          151,
          158,
          72,
          173,
          190,
          110,
          75,
          211,
          159,
          185,
          71,
          98,
        ],
      },
      "locType": "Identity",
      "requesterAddress": Object {
        "address": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        "equals": [Function],
        "toKey": [Function],
        "type": "Polkadot",
      },
      "userIdentity": Object {
        "email": "john.doe@logion.network",
        "firstName": "John",
        "lastName": "Doe",
        "phoneNumber": "+1234",
      },
    }
  }
>
  <PolkadotFrame
    title={
      <FrameTitle
        iconId="polkadot_collection"
        text="Collection Item recorded data"
      />
    }
  >
    <CertificateItemDetails
      item={
        Object {
          "files": Array [],
          "id": Hash {
            "_bytes": Uint8Array [
              154,
              174,
              187,
              189,
              63,
              143,
              230,
              81,
              184,
              126,
              131,
              99,
              4,
              85,
              167,
              10,
              67,
              150,
              76,
              105,
              145,
              7,
              123,
              180,
              80,
              191,
              26,
              62,
              210,
              210,
              92,
              204,
            ],
          },
        }
      }
    />
  </PolkadotFrame>
  <div
    className="frame-container"
  >
    <Frame
      border="2px solid #3B6CF4"
      title={
        <FrameTitle
          text="Statement of Facts"
        />
      }
    >
      <StatementOfFactsRequestButton
        itemId={
          Hash {
            "_bytes": Uint8Array [
              154,
              174,
              187,
              189,
              63,
              143,
              230,
              81,
              184,
              126,
              131,
              99,
              4,
              85,
              167,
              10,
              67,
              150,
              76,
              105,
              145,
              7,
              123,
              180,
              80,
              191,
              26,
              62,
              210,
              210,
              92,
              204,
            ],
          }
        }
      />
    </Frame>
  </div>
</LocPane>
`;
